### PR TITLE
Attach synapse logs to failed Playwright test runs

### PR DESCRIPTION
**Pull request by t3chguy**
For https://github.com/vector-im/wat-internal/issues/37

---
This change is marked as an *internal change* (Task), so will not be included in the changelog.

### CODE REVIEW

File: playwright/plugins/homeserver/index.ts [hunk 1]
```diff
@@ -53,7 +53,7 @@ export interface StartHomeserverOpts {
 
 export interface Homeserver {
     start(opts: StartHomeserverOpts): Promise<HomeserverInstance>;
-    stop(): Promise<void>;
+    stop(): Promise<string[]>;
```
Comment (richvdh) [inline playwright/plugins/homeserver/index.ts:56]: documentation please!

```suggestion
    /**
     * Stop this test homeserver instance.
     *
     * @returns A list of fully-qualified path names for logfiles generated during this test run.
     */
    stop(): Promise<string[]>;
```

File: playwright/element-web-test.ts [hunk 2]
```diff
@@ -102,7 +103,7 @@ export const test = base.extend<
     },
 
     startHomeserverOpts: "default",
-    homeserver: async ({ request, startHomeserverOpts: opts }, use) => {
+    homeserver: async ({ request, startHomeserverOpts: opts }, use, testInfo) => {
```
Comment (richvdh) [inline playwright/element-web-test.ts:106]: shouldn't testInfo have a type annotation? (I'm surprised typescript doesn't complain)

**Review by richvdh — APPROVED** [review]
seems generally plausible. Please document your interfaces though, to save people having to reverse-engineer them!

File: playwright/plugins/homeserver/index.ts (hunk 1 above)
Comment (t3chguy) [reply playwright/plugins/homeserver/index.ts:56]:
```suggestion
    /**
     * Stop this test homeserver instance.
     *
     * @returns A list of relative path names for logfiles generated during this test run.
     */
    stop(): Promise<string[]>;
```

File: playwright/element-web-test.ts (hunk 2 above)
Comment (t3chguy) [reply playwright/element-web-test.ts:106]: No as the `extend` method is well-typed:

image

**Pull request merged**